src/datasources/balances-api/entities/zerion-balance.entity.spec.ts

```diff
@@ -0,0 +1,557 @@
+import {
+  zerionAttributesBuilder,
+  zerionBalanceBuilder,
+  zerionBalancesBuilder,
+  zerionFlagsBuilder,
+  zerionFungibleInfoBuilder,
+  zerionImplementationBuilder,
+  zerionQuantityBuilder,
+} from '@/datasources/balances-api/entities/__tests__/zerion-balance.entity.builder';
+import {
+  ZerionAttributesSchema,
+  ZerionBalanceSchema,
+  ZerionBalancesSchema,
+  ZerionFlagsSchema,
+  ZerionFungibleInfoSchema,
+  ZerionImplementationSchema,
+  ZerionQuantitySchema,
+} from '@/datasources/balances-api/entities/zerion-balance.entity';
+import { faker } from '@faker-js/faker';
+import { ZodError } from 'zod';
+
+describe('Zerion Balance Entity schemas', () => {
+  describe('ZerionBalancesSchema', () => {
+    it('should validate a ZerionBalances object', () => {
+      const zerionBalances = zerionBalancesBuilder().build();
+
+      const result = ZerionBalancesSchema.safeParse(zerionBalances);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not allow invalid', () => {
+      const zerionBalances = {
+        data: 'invalid',
+      };
+
+      const result = ZerionBalancesSchema.safeParse(zerionBalances);
+
+      expect(!result.success && result.error).toStrictEqual(
+        new ZodError([
+          {
+            code: 'invalid_type',
+            expected: 'array',
+            received: 'string',
+            path: ['data'],
+            message: 'Expected array, received string',
+          },
+        ]),
+      );
+    });
+  });
+  describe('ZerionBalanceSchema', () => {
+    it('should validate a ZerionBalance object', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+
+      const result = ZerionBalanceSchema.safeParse(zerionBalance);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should not allow an invalid type value', () => {
+      const zerionBalance = zerionBalanceBuilder().build();
+      // @ts-expect-error - type is expected to be a 'positions' literal
+      zerionBalance.type = 'invalid';
```

Comment:
What do you think about adding an `unknown` literal here if it throws?

Reply:
Added in [efadc24](https://github.com/safe-global/safe-client-gateway/pull/1637/commits/efadc242b984be3bbd2955dff58249aaf050779c)